### PR TITLE
Adopt Bundler gemspec defaults in the generated extension

### LINKED ISSUE #11: Backport to dev_tools the tooling added to solidus_static_content
`solidus_static_content` had a number of recent changes devoted to easing the development experience. We want to bring in anything that's both useful and generic enough to be good for all extensions.

### What to do

Make a list of things that should be ported here, open PRs accordingly.

- [x] junit formatter for better circleci output (#6)
- [x] bin/rails bin/rake bin/setup bin/update bin/console scripts, possibly also bin/server and bin/release (#35, [original PR](https://github.com/solidusio-contrib/solidus_static_content/pull/39/files))
- [ ] standardize the extension structure to use the full extension name as main namespace and bundlers gemspec defaults ([original PR](https://github.com/solidusio-contrib/solidus_static_content/pull/39/files))
- [x] move from selenium to apparition + puma for feature specs ([original PR](https://github.com/solidusio-contrib/solidus_static_content/pull/42/files), #22)

### PR DESCRIPTION
**Pull request by aldesantis**
Fixes #11.

## Summary

Adopts some more Bundler gemspec defaults in the gemspec of the generated extension.

## Checklist

- [x] I have structured the commits for clarity and conciseness.
- [ ] ~I have added relevant automated tests for this change.~
- [x] I have added an entry to the changelog for this change.

### CODE REVIEW

File: spec/features/create_extension_spec.rb [hunk 1]
```diff
@@ -13,7 +13,8 @@
   let(:gemspec_name) { "solidus_#{extension_name}.gemspec" }
   let(:tmp_path) { Pathname.new(ext_root).join('spec', 'tmp') }
   let(:install_path) { tmp_path.join("solidus_#{extension_name}") }
-  let(:command_failed) { Class.new(StandardError) }
+
+  class CommandFailed < StandardError; end
```
Comment (elia) [inline spec/features/create_extension_spec.rb]: @aldesantis qualche motivo specifico per questa modifica? 
Facendo così CommandFailed viene definito su Object.

_Essendo nelle spec non è un grosso problema, ma come era prima pure aveva senso._

Comment (aldesantis) [reply spec/features/create_extension_spec.rb]: @elia leads to better error messages, but also happy to revert to the previous one. I'm not too afraid of polluting the namespace here tbh, but let me know.

Comment (elia) [reply spec/features/create_extension_spec.rb]: @aldesantis we could pull the error class along with most helpers to a support/ helper module, including that in the spec we should both solve the problem and cleanup the file.

Comment (elia) [reply spec/features/create_extension_spec.rb]: I created an issue, so we can mark this as solved ✔️

**Pull request merged**